mathesar_ui/src/components/ColumnName.svelte

```diff
@@ -1,16 +1,16 @@
 <script lang="ts">
-  import type { Column } from '@mathesar/api/tables/columns';
+  import type { ProcessedColumn } from '@mathesar/stores/table-data/processedColumns';
   import { getColumnIconProps } from './cell/utils';
   import TypeIcon from './TypeIcon.svelte';
 
-  export let column: Column;
+  export let column: ProcessedColumn;
```

Comment:
@pavish I saw you branch `data_explorer_ui_2` and realised that this is the only place where it will have actual conflict. Please take a look at this and let me know if you want to take this approach or the one in your branch. 

Explicitly mentioning the different here: 

This PR
`export let column: ProcessedColumn;`

Your branch
`export let column: CellColumnLike & { name: string };`

PS: Not that we will need all the constraints data here

Reply:
We cannot use `ProcessedColumn` here because this component is common to Tables & Queries, and queries have a very different column structure compared to tables.

It would be better to extend the type or add additional props to pass down to `getColumnIconProps`